digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Api.scala

```diff
@@ -1868,4 +1868,20 @@ class Digiroad2Api(val roadLinkService: RoadLinkService,
       case _ => None
     }
   }
+
+  get("/roadAssociationName") {
+    roadLinkService.getAllPrivateRoadAssociationNames()
+  }
+
+  get("/fetchAssociationNames/:associationName") {
+    val associationName = params("associationName").toString
+    val privateRoadAssociations = roadLinkService.getPrivateRoadsByAssociationName(associationName)
+    privateRoadAssociations.map { value =>
+      Map("name" -> associationName,
```

Comment:
associationName should be mapped from  **roadLinkService.getPrivateRoadsByAssociationName**

value =>  could be repalce by case (linkId, roadName, municiaplity)